src/joserfc/_keys.py

```diff
@@ -120,6 +120,10 @@ def __iter__(self) -> t.Iterator[Key]:
     def __bool__(self) -> bool:
         return bool(self.keys)
 
+    def __eq__(self, other: t.Any) -> bool:
+        assert isinstance(other, KeySet)
```

Comment:
This will raise an `AssertionError` not return `False`.

I think it's usually preferred to avoid `assert` outside of tests because in certain circumstances it can be removed by optimization flags.

Reply:
Separately: `isinstance()` checks can sometimes conflict with ducktyping. [This answer](https://stackoverflow.com/a/9844427/845210) suggests returning `NotImplemented` to give the other object a chance to check its own notion of equality.

Reply:
The assert is used to fix mypy issues